### PR TITLE
Feature/170 commodity data

### PR DESCRIPTION
**Pull request by Hoodathunk**
1.  Updated entries in CommodityDefinitions.cs
2.  Added code to create a log entry when unknown / out-of-sync commodity(s) are found.
3.  Added code to include sending 'economies' and 'prohibited' station data in EDDN 'Commodity' message.
4.  Fixed error of performing API /profile query on 'Docked' and 'Location' events.
5.  API update request from VoiceAttack also retrieves market data, if docked.

### CODE REVIEW

File: EDDNResponder/EDDNResponder.cs [hunk 1]
```diff
@@ -218,13 +230,18 @@ private void sendCommodityInformation()
                     data.Add("timestamp", DateTime.Now.ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ssZ"));
                     data.Add("systemName", systemName);
                     data.Add("stationName", EDDI.Instance.CurrentStation.name);
+                    if (eddnEconomies.Count > 0)
+                        data.Add("economies", eddnEconomies);
```
Comment (richardbuckle) [inline EDDNResponder/EDDNResponder.cs]: Again braces please!

File: EDDNResponder/EDDNResponder.cs [hunk 2]
```diff
@@ -218,13 +230,18 @@ private void sendCommodityInformation()
...
                     data.Add("commodities", eddnCommodities);
+                    if (EDDI.Instance.CurrentStation.prohibited.Count > 0)
+                        data.Add("prohibited", EDDI.Instance.CurrentStation.prohibited);
```
Comment (richardbuckle) [inline EDDNResponder/EDDNResponder.cs]: Braces please!

File: CompanionAppService/CompanionAppService.cs [hunk 3]
```diff
@@ -637,23 +651,64 @@ public static List<Module> OutfittingFromProfile(dynamic json)
...
+        {
+            List<String> ProhibitedCommodities = new List<String>();
+
+            if (json["lastStarport"] != null && json["lastStarport"]["prohibited"] != null)
+            {
+                foreach (dynamic prohibitedcommodity in json["lastStarport"]["prohibited"])
+                {
+                    String pc = (string)prohibitedcommodity.Value;
+                    if (pc != null)
+                        ProhibitedCommodities.Add(pc);
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply CompanionAppService/CompanionAppService.cs]: Fixed

File: CompanionAppService/CompanionAppService.cs [hunk 4]
```diff
@@ -669,6 +724,19 @@ public static List<Commodity> CommoditiesFromProfile(dynamic json)
                     }
                     Commodity.StatusFlags = StatusFlags;
                     Commodities.Add(Commodity);
+
+                    if (eddiCommodity == null || eddiCommodity.EDName != Commodity.EDName || eddiCommodity.name != Commodity.name
+                        || eddiCommodity.category != Commodity.category)
+                    {
+                        if (eddiCommodity.name != "Limpet")
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply CompanionAppService/CompanionAppService.cs]: Fixed

File: DataDefinitions/CommodityDefinitions.cs [hunk 5]
```diff
@@ -14,7 +14,7 @@ public class CommodityDefinitions
             {"atmosphericextractors", "atmosphericprocessors"},
             {"basicnarcotics", "narcotics"},
             {"hazardousenvironmentsuits", "hesuits"},
-            {"heliostaticfurnaces", "microbialfurnaces"},
+            //{"heliostaticfurnaces", "microbialfurnaces"},
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply DataDefinitions/CommodityDefinitions.cs]: I'm reluctant to lose any history from jgm's prior work, but I don't feel that strongly about it.

File: DataDefinitions/CommodityDefinitions.cs [hunk 6]
```diff
@@ -111,31 +111,31 @@ public class CommodityDefinitions
...
             {128671118, new Commodity(97, "Osmium", "Osmium", "Metals", 7591, false) },
-            {128671443, new Commodity(98, "S A P8 Core Container", "Sap 8 Core Container", "Salvage", 59196, false) },
-            {128671444, new Commodity(99, "Trinkets Of Fortune", "Trinkets Of Hidden Fortune", "Salvage", 1428, false) },
-            {128666754, new Commodity(100, "U S S Cargo Trade Data", "Trade Data", "Salvage", 2790, false) },
+            {128671443, new Commodity(98, "SAP8CoreContainer", "Sap 8 Core Container", "Salvage", 59196, false) },
+            {128671444, new Commodity(99, "TrinketsOfFortune", "Trinkets Of Hidden Fortune", "Salvage", 1428, false) },
+            {128666754, new Commodity(100, "USSCargoTradeData", "Trade Data", "Salvage", 2790, false) },
             {128672308, new Commodity(101, "ThermalCoolingUnits", "Thermal Cooling Units", "Machinery", 256, false) },
-            {128672313, new Commodity(102, "Skimer Components", "Skimmer Components", "Machinery", 859, false) },
+            {128672313, new Commodity(102, "SkimerComponents", "Skimmer Components", "Machinery", 859, false) },
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply DataDefinitions/CommodityDefinitions.cs]: Yes, I'm sure. after I add my 'commodity checking' code, comparing the commodities coming from the API station market data with the commodity definitions, this one was flagged.

File: DataDefinitions/CommodityDefinitions.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (richardbuckle) [reply DataDefinitions/CommodityDefinitions.cs]: I would argue that there's no need to keep dead wood in the live code base now that we have modern source control systems, and as I say it does impose a burden on future maintainers.

File: DataDefinitions/CommodityDefinitions.cs [hunk 7]
```diff
@@ -166,123 +165,123 @@ public class CommodityDefinitions
...
-            {300000121, new Commodity(242, "Alya Body Soap", "Medicines", 8218, true) },
-            {300000122, new Commodity(243, "Vidavantian Lace", "Consumer Items", 12615, true) },
-            {300000123, new Commodity(244, "Jaques Quinentian Still", "consumer Items", 2108, true) },
-            {300000124, new Commodity(245, "Soontill Relics", "Consumer Items", 19885, true) },
-            {128668547, new Commodity(246, "Unknown Artifact", "Thargoid Sensor", "Salvage", 290190, false) },
+            {300000121, new Commodity(242, "AlyaBodySoap", "Alya Body Soap", "Medicines", 8218, true) },
+            {300000122, new Commodity(243, "VidavantianLace", "Vidavantian Lace", "Consumer Items", 12615, true) },
+            {300000123, new Commodity(244, "JaquesQuinentianStill", "Jaques Quinentian Still", "Consumer Items", 2108, true) },
+            {300000124, new Commodity(245, "SoontillRelics", "Soontill Relics", "Consumer Items", 19885, true) },
+            {128668547, new Commodity(246, "UnknownArtifact", "Unknown Artifact", "Salvage", 290190, false) },
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply DataDefinitions/CommodityDefinitions.cs]: Good catch

File: DataDefinitions/CommodityDefinitions.cs [hunk 8]
```diff
@@ -296,30 +295,30 @@ public class CommodityDefinitions
...
-            {300000143, new Commodity(264, "Master Chefs", "Slavery", 20590, true) },
-            {300000144, new Commodity(265, "Crystalline Spheres", "Consumer Items", 12216, true) },
+            {300000143, new Commodity(264, "MasterChefs", "Master Chefs","Slaves", 20590, true) },
+            {300000144, new Commodity(265, "CrystallineSpheres", "Crystalline Spheres", "Consumer Items", 12216, true) },
             {128672775, new Commodity(266, "Taaffeite", "Taaffeite", "Minerals", 20696, false) },
             {128672776, new Commodity(267, "Jadeite", "Jadeite", "Minerals", 13474, false) },
             {128672810, new Commodity(268, "UnstableDataCore", "Unstable Data Core", "Salvage", 2427, false) },
-            {300000148, new Commodity(269, "Onionhead Alpha Strain", "Legal Drugs", 8437, true) },
-            {128672811, new Commodity(270, "DamagedEscapePod", "Occupied Escape Pod", "Salvage", 4474, false) },
+            {300000148, new Commodity(269, "Onionhead Alpha Strain", "Narcotics", 8437, true) },
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply DataDefinitions/CommodityDefinitions.cs]: Fixed

File: DataDefinitions/CommodityDefinitions.cs [hunk 9]
```diff
@@ -296,30 +295,30 @@ public class CommodityDefinitions
...
+            {300000150, new Commodity(272, "OnionheadBetaStrain", "Onionhead Beta Strain", "Narcotics", 8437, true) },
             {128673845, new Commodity(273, "Praseodymium", "Praseodymium", "Metals", 7156, false) },
             {128673846, new Commodity(274, "Bromellite", "Bromellite", "Minerals", 7062, false) },
             {128673847, new Commodity(275, "Samarium", "Samarium", "Metals", 6330, false) },
-            {128673848, new Commodity(276, "LowTemperatureDiamond", "Low Temperature Diamond", "Minerals", 57445, false) },
+            {128673848, new Commodity(276, "LowTemperatureDiamond", "Low Temperature Diamonds", "Minerals", 57445, false) },
             {128673850, new Commodity(277, "HydrogenPeroxide", "Hydrogen Peroxide", "Chemicals", 917, false) },
-            {128673851, new Commodity(278, "LiquidOxygen", "Liquid Oxygen", "Chemicals", 263, false) },
-            {128673852, new Commodity(279, "MethanolMonohydrateCrystals", "Methanol Monohydrate", "Minerals", 2282, false) },
+            {128673851, new Commodity(278, "LiquidOxygen", "Liquid oxygen", "Chemicals", 263, false) },
```
[the remark this answers is not in the document]
Comment (Hoodathunk) [reply DataDefinitions/CommodityDefinitions.cs]: Another one caught by my code

File: EDDNResponder/EDDNResponder.cs (hunk 1 above)
Comment (Hoodathunk) [reply EDDNResponder/EDDNResponder.cs]: Fixed

File: EDDNResponder/EDDNResponder.cs (hunk 2 above)
Comment (Hoodathunk) [reply EDDNResponder/EDDNResponder.cs]: Fixed

**Pull request merged**